GTMAppAuthSwift/Tests/GTMAppAuthFetcherAuthorizationTests.swift

```diff
@@ -283,4 +310,104 @@ class GTMAppAuthFetcherAuthorizationTest: XCTestCase {
     XCTAssertTrue(yAuthorization.userEmailIsVerified)
     XCTAssertTrue(yesAuthorization.userEmailIsVerified)
   }
+
+  // MARK: - Keychain Tests
+
+  func testSaveAuthorization() throws {
```

Comment:
The save and remove test cases still seem weird to me. We only test the `useDataProtectionKeychain` property is set or not. This is not enough to verify these methods. It is possible that the property is set and the keychain is not invoked at all. We need to check if the data is in the fake keychain assuming the fake keychain implementation is correct.
I understand it is kind of testing the KeychainHelperFake implementation. But this is also necessary. If the KeychainHelperFake implementation has some problems we can not trust these tests either.

Reply:
I understand your point of view. It is a little weird. My thinking is that the read, save, and remove methods would throw if things did not work (I have tests here demonstrating this behavior). Otherwise, if we do not trust that, then we will essentially be testing that the `Dictionary` methods for subscripting (AKA the read methods), `updateValue(_:forKey:)`, and `removeValue(forKey:)` work. That doesn't feel necessary.

The reason why I do check `keychainHelper.useDataProtectionKeychain` is that this value is our responsibility to set, and I want to make sure that our implementation does this (because it is used in the process of archiving later down the road). 

Last, I'll point out that the `testReadAuthorization()` (and others) method seems to do what you want. It saves an authorization, and then reads it. The read would not work if the save didn't occur.

Finally, I think another reason to not examine the `passwordStore` is that this is just an implementation detail of the fake. Testing it doesn't really build confidence, IMO. Further, if the implementation details of this fake ever changes to no longer have a `passwordStore`, which is reasonable and possible, then the tests would have to change (if we wrote tests against that dictionary). That's part of the reason why I've stuck to writing tests that exercise the API.